modules/imgcodecs/src/grfmt_png.cpp

```diff
@@ -505,7 +506,14 @@ bool  PngDecoder::readData( Mat& img )
                 }
 
                 memcpy(&m_chunkIHDR.p[8], &chunk.p[12], 8);
-                return true;
+
+                if (m_is_fcTL_loaded)
+                    return true;
+                else
+                {
+                    m_is_fcTL_loaded = true;
+                    return readData(img);
```

Comment:
The line changes simple imread behavior, right? I propose to return still image in case if user reads animation as single frame. It's APNG design decision and also presumes current OpenCV behaviour.